crates/router/src/core/payment_methods/cards.rs

```diff
@@ -314,44 +346,34 @@ pub async fn get_payment_method_from_hs_locker<'a>(
             .payload
             .get_required_value("RetrieveCardRespPayload")
             .change_context(errors::VaultError::FetchPaymentMethodFailed)?;
-        retrieve_card_resp
+        let enc_card_data = retrieve_card_resp
             .enc_card_data
             .get_required_value("enc_card_data")
-            .change_context(errors::VaultError::FetchPaymentMethodFailed)
+            .change_context(errors::VaultError::FetchPaymentMethodFailed)?;
+        decode_and_decrypt_locker_data(key_store, enc_card_data.peek().to_string()).await?
```

Comment:
Why are we using the key store to decrypt this data?
The key store based encryption is used for database-domain layer encryption and not for cards. Can you provide me with a bit more insight on this?

Reply:
This approach was decided earlier; out of scope for this change

cc: @jagan-jaya